kythe/go/indexer/emit.go

```diff
@@ -983,6 +984,29 @@ func (e *emitter) writeDoc(comments *ast.CommentGroup, target *spb.VName) {
 	e.writeFact(docNode, facts.NodeKind, nodes.Doc)
 	e.writeFact(docNode, facts.Text, strings.Join(lines, "\n"))
 	e.writeEdge(docNode, target, edges.Documents)
+	e.handleDeprecation(target, lines)
+}
+
+// handleDeprecation emits a deprecated fact for the specified target if the
+// comment lines indicate it is deprecated.
+func (e *emitter) handleDeprecation(target *spb.VName, lines []string) {
```

Comment:
nit: The rest of this package uses `writeXX` for functions that explicitly write facts/edges for a particular construct, or `emitXX` for functions that handle a general category of attributes (e.g., overrides) that may or may not be present. By that logic this is probably `emitDeprecation`, since it may or may not be present.

Reply:
Thanks, I missed (and was hoping to find) that distinction.

Reply:
Unfortunately it's not written down anywhere, and I may not have applied it entirely consistently either.